### ISSUE TITLE
Encoding Dropbox-API-Arg header

### ISSUE THREAD

**Issue by LucianoBestia** [dropbox/dropbox-sdk-rust#65]
# Encoding Dropbox-API-Arg header

I use the ureq default http client on dropbox-sdk version 0.13.1.

I created a file in my dropbox:  
/troškovi.txt  
If I try to download it with `dropbox_sdk::files::download()`.  
I got a "bad header" error. The filename contains a non-ASCII character and it is not encoded properly.  

In the documentation for Dropbox API v2
https://www.dropbox.com/developers/reference/json-encoding  
is described the  
"Encoding for JSON Arguments"  
for the Dropbox-API-Arg header:  

> If you use the Dropbox-API-Arg header, you need to make it "HTTP header safe". This means using JSON-style "\uXXXX" escape codes for the character 0x7F and all non-ASCII characters.

>We based this determination on the original HTTP/1.1 specification (RFC 2616) and as well as the newer HTTP/1.1 specification (RFC 7230). The former suggests that all you need to escape is 0x7F. The later recommends escaping all non-ASCII characters.

In default_client.rs line 252 I found this code:  
``` rust
Style::Upload | Style::Download => {
    // Send params in a header.
    req = req.set("Dropbox-API-Arg", params);
```
Is this a good place to encode the Dropbox-API-Arg header?  
Something like this:
``` rust
Style::Upload | Style::Download => {
    // Send params in a header.
    let encoded_params = encode_non_ascii(params);
    req = req.set("Dropbox-API-Arg", &encoded_params);    
```

``` rust
/// Encoding for JSON Arguments
/// for the Dropbox-API-Arg header
/// https://www.dropbox.com/developers/reference/json-encoding  
pub fn encode_non_ascii(params: &str)->String{
    let mut encoded_params = String::new();
    for ch in params.chars(){
        if ch.is_ascii(){
            encoded_params.push(ch);
        }else{
            encoded_params.push_str(&format!("\\u{:04x}", ch as u32 ));
        }
    }
    // return
    encoded_params
}
```
I tried this solution and it works. I just don't know if the placement in code is correct.

**Comment by wfraser** [comment]
Oh wow, this is actually the `ureq` header code raising this error, it's not even getting to Dropbox servers (bit of a sharp edge on that API, oof). Great bug report, you're completely right! I'll check the other headers we set too. Path-Root probably needs this same fix as well.

**Comment by wfraser** [comment]
fix is in v0.13.2

Thanks for reporting this!
